spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -69,15 +73,18 @@
  *
  * @since 1.1
  */
-public class PubSubSubscriberTemplate implements PubSubSubscriberOperations, DisposableBean {
+public class PubSubSubscriberTemplate
+		implements PubSubSubscriberOperations, DisposableBean {
 
 	private final SubscriberFactory subscriberFactory;
 
 	private final SubscriberStub subscriberStub;
 
 	private PubSubMessageConverter pubSubMessageConverter = new SimplePubSubMessageConverter();
 
-	private ExecutorService ackExecutor = Executors.newSingleThreadExecutor();
+	private ExecutorService defaultAckExecutor = Executors.newSingleThreadExecutor();
+
+	private Executor customAckExecutor;
```

Comment:
I'm fully believe that you can assign a `defaultAckExecutor` to this property by default and don't have that ternary logic in the `ack()` method.

Reply:
Yes! That's a nice idea.